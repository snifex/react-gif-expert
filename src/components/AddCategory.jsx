import { useState } from "react";
import PropTypes from 'prop-types';

export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState();

    const onInputChange = ({ target }) => {
        setInputValue( target.value );
    };

    const onSubmit = ( event ) => {
        event.preventDefault();
        const valueParsed = inputValue.trim()

        if( valueParsed.length <= 1 ){
            return;
        }

        //Al momento que recibimos la funciÃ³n del hook podemos hacer un callback
        //Donde tenemos los estados del hook anteriores en este caso categories
        setInputValue( "" );
        onNewCategory( valueParsed )
    };

    return (
        <form onSubmit={ (event) => onSubmit(event)} aria-label="form">
            <input 
                type="text" 
                placeholder="Buscar gifs"
                value={ inputValue }
                onChange={ onInputChange }
            />
        </form>

    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}