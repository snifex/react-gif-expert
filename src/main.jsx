import React from 'react';
import ReactDOM from 'react-dom/client';
import { GifExpertApp } from './GifExpertApp';
import './styles.css';

//Apikey = r2XsZAGITYZ07XPlqGzkNmqLGy8P0qEk

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <GifExpertApp />
  </React.StrictMode>
)
