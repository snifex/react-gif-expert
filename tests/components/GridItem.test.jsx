import { GifItem } from "../../src/components";
import { fireEvent, render, screen } from "@testing-library/react";

describe('Testing <GifItem/> component', () => { 

    const title = 'Saitama';
    const url = "https://one-punch.com/saitama.jpg"
    
    test('should match with the snapshot', () => {  
        const { container } = render( 
            <GifItem 
                title={title}
                url={url}
            />
        );
        expect( container ).toMatchSnapshot();
    });

    test('should show the image with the url and the alt text right', () => { 
        render( 
            <GifItem 
                title={title}
                url={url}
            />
        );

        //Sacamos los datos de la imagen desestructurandolo
        const { src, alt } =  screen.getByRole('img');

        expect( src ).toBe(url);
        expect( alt ).toBe(title);
    });

    test('should show the title in the component', () => { 
        render( 
            <GifItem 
                title={title}
                url={url}
            />
        );

        expect(screen.getByText(title)).toBeTruthy();
    })
});