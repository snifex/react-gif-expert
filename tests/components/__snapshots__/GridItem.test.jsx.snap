// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <GifItem/> component should match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://one-punch.com/saitama.jpg"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
